File: 2233-number-of-smooth-descent-periods-of-a-stock/number-of-smooth-descent-periods-of-a-stock.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        long result=1;
        int left=0;
        for(int right=1;right<prices.length;right++){
            result+=1;
            if(prices[right-1]==prices[right]+1){
                result+=(right-left);
            }else{
                left=right;
            }
        }
        return result;
        
    }
}

File: 2233-number-of-smooth-descent-periods-of-a-stock/number-of-smooth-descent-periods-of-a-stock.py
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        result=1
        left=0
        for right in range(1,len(prices)):
            result+=1
            if prices[right-1]==prices[right]+1:
                result+=(right-left)        
            else:
                left=right
        return result